Allow system wlroots to reduce maintenance (wlroots has tons of options)

--- meson.build.orig	2022-11-07 12:16:30 UTC
+++ meson.build
@@ -37,8 +37,8 @@ add_project_arguments(
   ],
   language: 'cpp')
 
-wlroots = subproject('wlroots', default_options: ['examples=false'])
-have_xwlr = wlroots.get_variable('features').get('xwayland')
+wlroots = dependency('wlroots', version: ['>=0.16.0', '<0.17.0'], default_options: ['examples=false'])
+have_xwlr = wlroots.get_variable(pkgconfig: 'have_xwayland', internal: 'have_xwayland') == 'true'
 xcb_dep = dependency('xcb', required: get_option('xwayland'))
 
 if get_option('xwayland').enabled() and not have_xwlr
--- src/meson.build.orig	2022-11-07 12:16:30 UTC
+++ src/meson.build
@@ -7,7 +7,7 @@ executable('Hyprland', src,
     server_protos,
     dependency('wayland-server'),
     dependency('wayland-client'),
-    wlroots.get_variable('wlroots'),
+    wlroots,
     dependency('cairo'),
     dependency('pango'),
     dependency('pangocairo'),
